core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.ts

```diff
@@ -84,30 +143,157 @@ export class ContributorAdminDashboardPageComponent implements OnInit {
+        response => {
+          this.topics = response;
+          response.map((
+              topic) =>
+            this.allTopicNames.push(topic.topic));
+          this.applyTopicFilter();
+        }
+      );
+  }
+
+  toggleLanguageDropdown(): void {
+    this.languageDropdownShown = !this.languageDropdownShown;
+  }
+
+  toggleActivityDropdown(): void {
+    this.activityDropdownShown = !this.activityDropdownShown;
+  }
+
+  applyTopicFilter(): void {
+    this.selectedTopics.ids = this.selectedTopics.topics.map(
+      selectedTopic => {
+        const matchingTopic = this.topics.find(
+          topicChoice => topicChoice.topic === selectedTopic);
+        return matchingTopic ? matchingTopic.id : '';
+      });
+    this.filter = new ContributorAdminDashboardFilter(
+      this.selectedTopics.ids,
+      this.selectedLanguage.id,
+      null,
+      this.selectedLastActivity);
+  }
+
+  selectLanguage(language: string): void {
+    const currentOption: LanguageChoice = this.languageChoices.find(
+      option => option.language === language) || {
+      language: 'English',
+      id: 'en'
+    };
+    this.selectedLanguage.language = currentOption.language;
+    this.selectedLanguage.id = currentOption.id;
+    this.translationReviewersCount = this.translationReviewersCountByLanguage[
+      this.selectedLanguage.id];
+    this.filter = new ContributorAdminDashboardFilter(
+      this.selectedTopics.ids,
+      this.selectedLanguage.id,
+      null,
+      this.selectedLastActivity);
+    this.languageDropdownShown = false;
+  }
+
+  selectLastActivity(lastActive: number): void {
+    if (lastActive === 0) {
+      this.selectedLastActivity = lastActive;
+      this.filter = new ContributorAdminDashboardFilter(
+        this.selectedTopics.ids,
+        this.selectedLanguage.id,
+        null);
+      this.activityDropdownShown = false;
+      return;
+    }
+    this.selectedLastActivity = lastActive;
+    this.filter = new ContributorAdminDashboardFilter(
+      this.selectedTopics.ids,
+      this.selectedLanguage.id,
+      null,
+      this.selectedLastActivity);
+    this.activityDropdownShown = false;
+  }
+
+  setActiveTab(tabName: string): void {
+    this.filter = new ContributorAdminDashboardFilter(
+      this.selectedTopics.ids,
+      this.selectedLanguage.id,
+      null,
+      this.selectedLastActivity);
+    this.activeTab = tabName;
+  }
+
+  checkMobileView(): boolean {
+    return (this.windowRef.nativeWindow.innerWidth < 800);
+  }
+
+
+  updateSelectedContributionType(selectedContributionType: string): void {
+    this.selectedContributionType = selectedContributionType;
+    this.setActiveTab(selectedContributionType);
+  }
+
+  /**
+  * Close dropdown when outside elements are clicked
+  * @param event mouse click event
+  */
+  @HostListener('document:click', ['$event'])
+  onDocumentClick(event: MouseEvent): void {
+    const targetElement = event.target as HTMLElement;
+    if (!this.checkMobileView()) {
```

Comment:
Flip this conditional into a guard clause to reduce nesting
```suggestion
    if (this.checkMobileView()) {
      return;
    }
    ...
```

Reply:
Done. Thanks.